### PR TITLE
Feature/test clean up

### PR DESCRIPTION
**Pull request by mrMakaronka**
Fix: #91, #109

### CODE REVIEW

File: runtime/src/main/java/com/spbsu/flamestream/runtime/graph/materialization/RouterJoba.java [hunk 1]
```diff
@@ -22,29 +24,47 @@
...
 
-  public RouterJoba(ComputationProps layout,
+  private Joba localJoba;
+
+  public RouterJoba(String nodeId,
+                    ComputationProps layout,
                     Map<String, ActorRef> managerRefs,
                     HashFunction hashFunction,
                     GraphManager.Destination destination,
+                    ActorRef acker,
```
Comment (marnikitta) [inline runtime/src/main/java/com/spbsu/flamestream/runtime/graph/materialization/RouterJoba.java]: 7 args... Can we replace `layout + managerRef` with built `IntRangeMap`. And `nodeId `with `localRange`. So there is no information about "workers", their names in the router.

File: runtime/src/test/java/com/spbsu/flamestream/runtime/FilterAcceptanceTest.java [hunk 2]
```diff
@@ -39,28 +39,34 @@ private static Graph multiGraph() {
...
+                        .ints(streamSize)
+                        .peek(value -> expected.add(value * -1 * -2 * -3 * -4))
+                        .boxed())
+                .limit(parallelism).collect(Collectors.toList());
 
-    final Consumer<Object> randomConsumer = randomConsumer(
-            flame.attachFront("linearFilterFront", new AkkaFrontType<>(runtime.system()))
-                    .collect(Collectors.toList())
-    );
+        final AwaitConsumer<Integer> consumer = new AwaitConsumer<>(streamSize * parallelism);
```
Comment (marnikitta) [inline runtime/src/test/java/com/spbsu/flamestream/runtime/FilterAcceptanceTest.java]: It is confusing that test on business logic depends on "runtime specific" param.

File: examples/src/test/java/com/spbsu/flamestream/example/wordcount/WordCountTest.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+/**
+ * User: Artem
+ * Date: 19.12.2017
+ */
+public class WordCountTest extends FlameAkkaSuite {
+  @Test
+  public void localEnvironmentTest() throws InterruptedException {
+    final int parallelism = 4;
+    try (final LocalRuntime runtime = new LocalRuntime(parallelism, 50)) {
```
[the remark this answers is not in the document]
Comment (marnikitta) [reply examples/src/test/java/com/spbsu/flamestream/example/wordcount/WordCountTest.java]: Also let make `parallelism = 4` default, as there is no other parallelism in our tests)

File: runtime/src/test/java/com/spbsu/flamestream/runtime/FlameAkkaSuite.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.List;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * User: Artem
+ * Date: 22.12.2017
+ */
+public class FlameAkkaSuite extends FlameStreamSuite {
+  protected <T> void applyDataToHandles(List<Stream<T>> data, List<AkkaFrontType.Handle<T>> handles) {
```
[the remark this answers is not in the document]
Comment (marnikitta) [reply runtime/src/test/java/com/spbsu/flamestream/runtime/FlameAkkaSuite.java]: Resolved

File: examples/src/test/java/com/spbsu/flamestream/example/wordcount/WordCountTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mrMakaronka) [reply examples/src/test/java/com/spbsu/flamestream/example/wordcount/WordCountTest.java]: 100 for each source is 400 in total. It is too much.

File: runtime/src/main/java/com/spbsu/flamestream/runtime/edge/FrontActor.java [hunk 5]
```diff
@@ -14,9 +14,9 @@
   private final Front front;
 
   private FrontActor(EdgeContext context, FlameRuntime.FrontInstance<?> frontInstance) throws
-                                                                          IllegalAccessException,
-                                                                          InvocationTargetException,
-                                                                          InstantiationException {
+          IllegalAccessException,
```
[the remark this answers is not in the document]
Comment (mrMakaronka) [reply runtime/src/main/java/com/spbsu/flamestream/runtime/edge/FrontActor.java]: 👍

File: examples/src/test/java/com/spbsu/flamestream/example/index/InvertedIndexTest.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    };
+  }
+
+  @Test(dataProvider = "provider")
+  public void test(InvertedIndexValidator validator, boolean backPressure) throws InterruptedException {
+    try (final LocalRuntime runtime = new LocalRuntime(4, 10)) {
+      final FlameRuntime.Flame flame = runtime.run(new InvertedIndexGraph().get());
+      {
+        final AwaitConsumer<WordBase> awaitConsumer = new AwaitConsumer<>(validator.expectedOutputSize());
+        flame.attachRear("Rear", new AkkaRearType<>(runtime.system(), WordBase.class))
```
[the remark this answers is not in the document]
Comment (mrMakaronka) [reply examples/src/test/java/com/spbsu/flamestream/example/index/InvertedIndexTest.java]: Fixed

File: runtime/src/main/java/com/spbsu/flamestream/runtime/graph/materialization/RouterJoba.java (hunk 1 above)
Comment (mrMakaronka) [reply runtime/src/main/java/com/spbsu/flamestream/runtime/graph/materialization/RouterJoba.java]: Fixed

File: runtime/src/test/java/com/spbsu/flamestream/runtime/FilterAcceptanceTest.java (hunk 2 above)
Comment (mrMakaronka) [reply runtime/src/test/java/com/spbsu/flamestream/runtime/FilterAcceptanceTest.java]: Fixed

File: examples/src/test/java/com/spbsu/flamestream/example/index/InvertedIndexTest.java [hunk 7]
```diff
@@ -34,7 +34,7 @@
 
   @Test(dataProvider = "provider")
   public void test(InvertedIndexValidator validator, boolean backPressure) throws InterruptedException {
-    try (final LocalRuntime runtime = new LocalRuntime(4, 10)) {
+    try (final LocalRuntime runtime = new LocalRuntime(DEFAULT_PARALLELISM, 10)) {
```
Comment (marnikitta) [inline examples/src/test/java/com/spbsu/flamestream/example/index/InvertedIndexTest.java]: Actually, I mean constructor with no arguments, but this seems to be fine too)

File: runtime/src/main/java/com/spbsu/flamestream/runtime/utils/collections/ListIntRangeMap.java [hunk 8]
```diff
@@ -33,17 +36,40 @@ public T get(int key) {
...
+    mapping.addAll(map.entrySet());
+  }
+
+  @Override
+  public Set<Entry<T>> entrySet() {
+    return mapping;
   }
 
-  private static final class RangeEntry<T> {
+  public static final class Entry<T> {
```
Comment (marnikitta) [inline runtime/src/main/java/com/spbsu/flamestream/runtime/utils/collections/ListIntRangeMap.java]: This could actually implement Map.Entry interface

**Pull request merged**